prow/plugins/approve/approve.go

```diff
@@ -125,12 +125,32 @@ func helpProvider(config *plugins.Configuration, enabledRepos []string) (*plugin
 		}
 		approveConfig[repo] = fmt.Sprintf("Pull requests %s require an associated issue.<br>Pull request authors %s implicitly approve their own PRs.<br>The /lgtm [cancel] command(s) %s act as approval.<br>A GitHub approved or changes requested review %s act as approval or cancel respectively.", doNot(opts.IssueRequired), doNot(opts.HasSelfApproval()), willNot(opts.LgtmActsAsApprove), willNot(opts.ConsiderReviewState()))
 	}
+
+	yamlSnippet, err := plugins.CommentMap.GenYaml(&plugins.Configuration{
+		Approve: []plugins.Approve{
+			{
+				Repos: []string{
+					"ORGANIZATION",
+					"ORGANIZATION/REPOSITORY",
+				},
+				IssueRequired:       false,
```

Comment:
I'm having a hard time understanding how this works -- why do we need to add values here? For "generic" looking examples?

Reply:
Yes, these are sample values for the *specific* plugin. It is a way to have control over the values we want to display in the example snippets in UI. The fields are *explicitly defined* while the comments are added *dynamically* via [genyaml](https://github.com/kubernetes/test-infra/tree/master/pkg/genyaml). What are your thoughts on this implementation?

![image](https://user-images.githubusercontent.com/9957358/64446100-d3dd3800-d08c-11e9-8560-69bb539ac67f.png)

Reply:
I think for types where we have something semantically meaningful to put into the example (like `"ORGANIZATION"`) I agree 100% with you -- for all the fields that are simple types and the values are their zero value or similar, though, couldn't we just default those? Do we want to hide valid configuration fields in the examples at any point? If not, maybe a default that the user could overwrite with a more meaningful one could be useful to make sure all fields get shown and adding new fields is simpler.

Reply:
> I agree 100% with you -- for all the fields that are simple types and the values are their zero value or similar, though, couldn't we just default those? 
That seems reasonable to me for primitive types.

> Do we want to hide valid configuration fields in the examples at any point? 
Not that I am aware of. I would presume the `default` example would include **all** options available. Conversely, there *may* be a use-case where we want *multiple* example snippets - possibly to configurations *common* configurations to achieve a specific outcome. 

> If not, maybe a default that the user could overwrite with a more meaningful one could be useful to make sure all fields get shown and adding new fields is simpler.
I am not sure I understand. Who is the user here? An organization? If there is a need to customize this then we can treat is similar to the `Config` where we have a snippet for each repo. The one caveat being that the payload response will be huge as we add more examples.

Reply:
@stevekuznetsov - I updated to *include* all options, while *excluding* primitive types (with non-nil defaults with same value as what was specified) in call to `GenYaml`.

Can you clarify your last point?

> > If not, maybe a default that the user could overwrite with a more meaningful one could be useful to make sure all fields get shown and adding new fields is simpler.
> 
> I am not sure I understand. Who is the user here? An org/repo? If there is a need to customize this then we can treat is similar to the `Config` where we have a snippet for each repo. The one caveat being that the payload response will be huge as we add more examples.

Reply:
Sorry, I meant to generate defaults but allow a config author when writing documentation to overwrite the generated ones with more semantically meaningful snippets.